src/hooks/useTransactionPreparation.ts

```diff
@@ -77,16 +78,17 @@ export default function useTransactionPreparation({
       const estimatedFeeType = sourceApi.registry.createType('Option<Balance>', estimatedFeeCall);
       const estimatedFee = estimatedFeeType.toString();
 
-      dispatchTransaction(TransactionActionCreators.estimateFee(estimatedFee));
+      dispatchTransaction(TransactionActionCreators.setEstimateFee(estimatedFee));
     };
 
-    if (areApiReady && payload) {
+    if (areApiReady && payload && !globalEstimatedFee) {
```

Comment:
No matching should be necessary - if this `useEffect` is fired too many times it means that some dependency is changing, ergo we need to figure out which dependency and why.

Reply:
The loops starts because the `Transfer` component gets updated because of `estimatedFee` global value update, the this forces the child components to rerender and that's where the loop starts.
I think that adding the match is enough to solve this since this prevents the rerenders due to the issue described above.

Lastly if this answer does not satisfy your requirements , i would ask you to see this as a workaround for fixing the deployed UI and solve this during refactor process ( which i'm sure this process will be revisited )


Anyway i see this as a workaround until we can start with the refactor that will eliminate all this .

Reply:
Re-rendering should not be an issue - if we render the same components, they call the same hooks the `useEffect` inner function should not be called in case none of the dependencies have changed.
What you are doing here is basically a hack and even given architecture refactoring around the corner I'm against introducing anything like that.

>  which i'm sure this process will be revisited 
What if it doesn't? I really think we must at least add an issue and a comment here to make sure this gets addressed in the future. Adding hacks to code without even marking them as such is imho a huge no-go.

I find it hard to believe that it would take much time to dig a little bit deeper to figure out which of these variables:
```
[
    areApiReady,
    dispatchTransaction,
    estimatedFeeMethodName,
    laneId,
    payload,
    sourceApi.registry,
    sourceApi.rpc.state,
    targetChain
  ]
```
changes between re-renders. We already know it's not `payload`, cause comparing it with `prevPayload` solves the issue.
For a start and a naive method, it might be just a matter of adding `usePrevious` for each one of them and console.logging the shallow comparison results.

Reply:
super weird .. it seems the one making the useEffect to enter in that loop is the `areApiReady` coming from the hook `useLoadingApi` ... i'm trying to understand why this is happening ... below the useLoadingApi hook code: 


```
export default function useLoadingApi(): LoadingStates {
  const {
    sourceChainDetails: {
      apiConnection: { isApiReady: isSourceApiReady }
    },
    targetChainDetails: {
      apiConnection: { isApiReady: isTargetApiReady }
    }
  } = useSourceTarget();

  return {
    areApiReady: isSourceApiReady && isTargetApiReady,
    sourceReady: isSourceApiReady,
    targetReady: isTargetApiReady
  };
}
```

Reply:
i think i found it .... `useLoadingApi` hook retrieves the object : 

```
{
    areApiReady: isSourceApiReady && isTargetApiReady,
    sourceReady: isSourceApiReady,
    targetReady: isTargetApiReady 
}
```

and the calculateFee useEffect was using `const areApiReady = useLoadingApi();` meaning that instead of checking `areApiReady.areApiReady` ( boolean ) we were checking `areApiReady` which is an object ( object that is different in each rerender ) ... 

anyway ... super peaky ... thanks @tomusdrw  for pushing me to make the things right. Honestly i thought it was something more related to architecture.

Reply:
Nice! That's an awesome finding, thanks for taking the time to identify something that we missed in the review :) As all of these kind of bugs in hindsight it looks so obvious :)